### PR TITLE
chore: add refund tests based on spec conditions

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

**Comment by gregdhill** [comment]
I think these are the main conditions that touch this pallet exclusively, not sure what to do about the other conditions that rely on other modules.

File: crates/refund/src/tests.rs [hunk 1]
```diff
@@ -55,3 +55,59 @@ fn test_refund_succeeds() {
...
+                .mock_safe(|_, _, _, _, _| MockResult::Return(Ok(())));
+            ext::vault_registry::try_increase_to_be_issued_tokens::<Test>.mock_safe(|_, _| MockResult::Return(Ok(())));
+            ext::vault_registry::issue_tokens::<Test>.mock_safe(|_, _| MockResult::Return(Ok(())));
+
+            let redeem_id = H256::random();
+            <RefundRequests<Test>>::insert(redeem_id, RefundRequest { ..Default::default() });
+
+            assert_ok!(Refund::_execute_refund(redeem_id, vec![0u8; 100], vec![0u8; 100]));
+
+            assert!(
```
Comment (sander2) [inline crates/refund/src/tests.rs:107]: You're only testing 1 out of the 4 post conditions?

Comment (sander2) [reply crates/refund/src/tests.rs:107]: Also leaving out a ton of the preconditions, but probably that is ok - the post conditions are most important

Comment (gregdhill) [reply crates/refund/src/tests.rs:107]: As commented above, most of the other conditions are directly related to the functionality of other components so I'm not sure it makes sense to add here.

Comment (sander2) [reply crates/refund/src/tests.rs:107]: Well, the ticket does say spec based _integration_ tests. I agree that in this unit test it wouldn't really make sense to check, but I would suggest making it an integration test instead

**Review by sander2 — COMMENTED** [review]
Do we ignore this postcondition?
"The :ref:`totalSupply` in the :ref:`treasury-module` MUST increase by fee."

**Comment by gregdhill** [comment]
> Do we ignore this postcondition?
> "The :ref:`totalSupply` in the :ref:`treasury-module` MUST increase by fee."

I can certainly add tests to ensure the rewards increase, but IMO we should reword this condition.

**Pull request merged**